src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -75,19 +77,29 @@ SGMatrix<float64_t> RandomFourierGaussPreproc::apply_to_matrix(SGMatrix<float64_
 	return projection;	
 }
 
-SGMatrix<float64_t> RandomFourierGaussPreproc::sample_spectral_density(int32_t dim_input_space) const
+SGMatrix<float64_t> RandomFourierGaussPreproc::sample_spectral_density(int32_t dim_input_space)
 {
-	NormalDistribution<float64_t> normal_dist;
-	auto sampled_kernel = SGMatrix<float64_t>(m_dim_output, dim_input_space);
-	const auto width = get_width();
-	const auto std_dev = std::sqrt(2.0 / width);
-	std::generate(
-	    sampled_kernel.begin(), sampled_kernel.end(),
-	    [this, &normal_dist, &std_dev]() {
-		    return std_dev * normal_dist(m_prng);
-	    });
-
-	return sampled_kernel;
+
+	SGMatrix<float64_t> sampled_kernel(m_dim_output, dim_input_space);
+
+	if (auto kernel = std::dynamic_pointer_cast<GaussianKernel>(m_kernel))
+	{
+		NormalDistribution<float64_t> normal_dist;
+		const auto width = kernel->get_width();
+		const auto std_dev = std::sqrt(2.0 / width);
+		std::generate(
+		    sampled_kernel.begin(), sampled_kernel.end(),
+		    [this, &normal_dist, &std_dev]() {
+			    return std_dev * normal_dist(m_prng);
+		    });
+
+		return sampled_kernel;
+	}
+	else
+	{
+		not_implemented(SOURCE_LOCATION);
+		return SGMatrix<float64_t>();
+	}
```

Comment:
could you move the return to the end of the function? Since you allocate the memory before the conditions you might as well have a single return statement